File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        """maxsum = nums[0]  # handle negatives
        for i in range(len(nums)):
            sums = 0
            for j in range(i, len(nums)):
                sums += nums[j]
                maxsum = max(maxsum, sums)
        return maxsum
        """
        cursum=0
        maxsum=nums[0]

        for i in range(len(nums)):
            cursum+=nums[i]
            maxsum=max(maxsum,cursum)
            if cursum<0:
                cursum=0
        return maxsum

File: 0053-maximum-subarray/0053-maximum-subarray.c
int max(int a, int b){
    return a>b? a:b;
}
int maxSubArray(int* nums, int numsSize) {
    if (numsSize == 0){return 0;
    }
    int max_so_far=nums[0];
    int max_ending_here=nums[0];
    for (int i=1 ; i<numsSize;i++){
        max_ending_here=max(nums[i],max_ending_here+nums[i]);
        max_so_far=max(max_so_far,max_ending_here);
    }
    return max_so_far;

    
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/leethub-v4/bcilpkkbokcopmabingnndookdogmbna